### DISCUSSION TITLE
Translation CLA

### DISCUSSION THREAD

**Discussion by comradekingu** [tenacityteam/tenacity-legacy#382] [General]
Some parts of https://hosted.weblate.org/contributor-agreement/tenacity/tenacity/?next=/projects/tenacity/tenacity/
are superfluous.

The license is covered in 6.4 of https://hosted.weblate.org/legal/terms/
E-mail/name covered in 5,4, 5.7 and 6.5 of https://hosted.weblate.org/legal/terms/

There is a similar document for GitHub, and a DCO is not needed.
I will not sign one for my contributions.

**Comment by Semisol** [comment]
The DCO is not a CLA.
By "signing" it, via `--signoff` or via other ways, you are saying:
1. I have rights to relicense this under GPL-2, or
2. I made this contribution and I am licensing it under GPL-2, or
3. someone else which certified 1, 2 or 3 provided me this contribution, and I have not modified it
4. and that you agree for a permanent record of your signoff being kept indefinitely 

Please check @emabrey's answer, she explains it in more detail.

**Comment by emabrey** [comment]
> 
> 
> Some parts of https://hosted.weblate.org/contributor-agreement/tenacity/tenacity/?next=/projects/tenacity/tenacity/
> are superfluous.
> 
> The license is covered in 6.4 of https://hosted.weblate.org/legal/terms/
> E-mail/name covered in 5.3, 5,4, 5.7 and 6.5 of https://hosted.weblate.org/legal/terms/
> 
> There is a similar document for GitHub, and a DCO is not needed.
> I will not sign one for my contributions.

1. The DCO we are using is literally the same one used by Linux.
2. It is not a CLA, the feature we used on Weblate simply calls it a CLA.
3. The term "provider" in the Weblate terms of use means `Ing. Michal Čihař, Nábřežní 694, 471 54 Cvikov, CZ, Reg. No. 04705904`, so any rights you may or may not have granted "the provider" has nothing to do with the DCO and nothing directly to do with contributing.
4. The rights you have granted "the provider" via the terms of use for Weblate have nothing to do with us.
5. The part that refers to "(including all personal information I submit with it, including my sign-off)" is literally just telling you that we require you to submit your name and e-mail as part of a git commit, and that if you do that, we can't reverse it due to how git works/the impracticality of scrubbing public information. It also means that if you accidentally put unwanted and unrequested information into a contribution, like say your home country or something other than name and email, we can't undo the fact that you made it public (even if it were an accident).

We just wanted to make it crystal clear that when you contribute on Weblate, you are signing off the commits that contain those changes just the same as if you had run a git commit with the changes inside of it. We wanted to make that clear because it will make your account email and account name from Weblate show up in the git commit history, and that isn't obvious to most people unless we explicitly tell you. To avoid accidents where people who aren't okay with that somehow end-up unknowingly agreeing to the DCO, we made it an unavoidable notice for Weblate contributors.

Basically, that document tells you that to make a Weblate contribution you have to provide the same information that everyone else who makes a git commit provides: a name (the one associated with your Weblate account already) and an email (the one associated with your Weblate account already). Since we are adding it to a git commit it will be public on the git repository and thus can be downloaded and distributed across a potentially unlimited number of computers (that's how version control like git works), meaning we can't reverse the process of signing off a commit.

Please see https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin for more information on what a DCO is and why it is needed.

**Reply by emabrey** [reply]
https://github.com/tenacityteam/tenacity/pull/383/commits/81175a0c97666156ed88714092e832c5ac667935 is an example of what it looks like over here on the git repository when you make a Weblate contribution.

**Reply by comradekingu** [reply]
@emabrey I don't dispute that is what it looks like. I already knew, because I read the Hosted Weblate terms when I agreed to them.
You presumably did not. I assume that is why the Tenacity translation CLA [has now changed](https://hosted.weblate.org/changes/?component=tenacity&project=tenacity&lang=&action=56&user=&start_date=&end_date=). Good.
Hopefully the seventh iteration will be its complete removal.

Whether it still makes for a "executed, binding agreement" or anything that constitutes a CLA between "any legal entity or an individual other than the Provider, who is using the Service" and Tenacity the project is neither here nor there.

I think it is superfluous, and much like Linux, your project doesn't have the DCO from any prior contributors.
The whole of Audacity did fine without one, as with every other project on Hosted Weblate.

You presumably didn't read line 2 and 3 or the DCO 1.1 in omitting

> Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
> 
> Everyone is permitted to copy and distribute verbatim copies of this license document, but changing it is not allowed.

There is a level of irony to committing the exact thing you are trying to prevent by the non-letter of its law.
Also the license was attemptedly changed to GPLv2-only https://hosted.weblate.org/changes/?component=tenacity&project=tenacity&lang=&action=55&user=&start_date=&end_date= which luckily was also only spam translators see because it was changed back with no contributions to its name.

To clarify:

https://weblate.org/en/terms/
6.4 The User agrees, upon contributing to a Project, to the license the Project has specified.

@Semisol
There is no other party the User can make an agreement with.
If that isn't good enough from the User to the Provider, why does this project use the Service between itself and the Hosted Weblate Provider?

This is the Tenacity GPLv2 license, which does not have a cure clause.

> 2. You may modify your copy or copies of the Program or any portion
> of it, thus forming a work based on the Program, and copy and
> distribute such modifications or work under the terms of Section 1
> above, provided that you also meet all of these conditions:

>     b) You must cause any work that you distribute or publish, that in
>     whole or in part contains or is derived from the Program or any
>     part thereof, to be licensed as a whole at no charge to all third
>     parties under the terms of this License.

So the license is the license, and is agreed to specifically.
The software can't be modified without granted merchantability,
which necessitates every copyright that can be exercised.

This does apply to Tenacity:

>But when you
distribute the same sections as part of a whole which is **a work based
on the Program**, the distribution of the whole must be on the terms of
this License, whose permissions for other licensees extend to the
entire whole, and thus to each and every part regardless of who wrote it.

>Each time you redistribute the Program (or any work based on the
Program), the recipient automatically receives a license from the
original licensor to copy, distribute or modify the Program subject to
these terms and conditions.  You may not impose any further
restrictions on the recipients' exercise of the rights granted herein.

The user would have to accept new Hosted Weblate terms to change the indefinite nature of name and e-mail in Git commits,
so point 4. is also void IMO.

**Reply by emabrey** [reply]
You are technically correct that the Weblate TOS covers the licensing issue, so you are correct that the DCO is technically superfluous in the sense of covering the legalities of the license. We do not need to remind you or any other Weblate contributor that you have agreed to the license of Tenacity when you contribute, that is already covered by the TOS of Weblate, as you rightfully pointed out. However, we included the DCO on Weblate to explicitly and openly remind Weblate contributors that they are sharing their name and their email on a different website (namely GitHub and Sourcehut, for example). That is a privacy concern, and we respect people's privacy. Relying on the TOS and not being extremely open about that information being part of the contribution isn't something we are interested in doing. It would be technically and legally okay to do that of course, you are right about that, but it wouldn't be fully open and transparent about what we do with your Weblate name and Weblate email, which isn't good. Additionally, the DCO is a way to make sure that if you or any other contributor break the terms of the GPL license and it gets me or another Tenacity developer in trouble with a 3rd-party, we can more easily track down who was responsible for promising us that they didn't violate the license. Without a DCO we can't assign blame for the failure to comply with the license, and the GPL by itself doesn't accomplish that assignment of blame, which I believe is something you failed to consider. That is the same reason Linux uses it. It doesn't modify the license, it is an additional safeguard within the same framework as the existing license.

**Comment by comradekingu** [comment]
And no, [registering the project again](https://hosted.weblate.org/changes/?project=&lang=&action=51&action=50&user=emabrey&start_date=&end_date=) doesn't clear the [CLA commit log](https://hosted.weblate.org/changes/?component=tenacity&project=tenacity&lang=&action=56&user=&start_date=&end_date=).
It does however clear the project contribution history, taking all contributions from the last ~2.5 weeks with it.
Either they are lost, or adding them after the fact does put them in accordance with the DCO.

**Reply by emabrey** [reply]
I wasn't trying to clear the Weblate history. You seem to have made a bunch of false assumptions. In fact I was trying to fix an issue with someone having force-pushed a bunch of things on our Github master and it broke the HEAD refs between the remote Weblate repository and the Tenacity repository. When I tried to manually merge it there was no way to recover it without being able to force push to Weblate, which you cannot do, or I would've had to manually do a gigantic merge that likely would've resulted in the corruption of the PO files if I made any mistakes. Additionally, all those "CLA" field log changes are because Weblate has a bug that results in you getting a "Detected as Spam!" warning about the "CLA" field anytime you update any setting on any of the settings pages for the component. After randomly deleting random parts of the "CLA" field (which defeats the spam bug) while I made multiple settings changes entirely unrelated to the "CLA" field, I eventually deleted it to blank since that is a workaround for the Weblate bug (it won't spam detect a blank "CLA" field) and then once the rest of the settings were in a usable state, I set the "CLA" field back to where it was always intended to be. I have no idea where you got all these false ideas about my motivations.

Also, I don't think there are any missing contributions (though they may be missing from the Weblate UI, that I don't know). The actual content of the contributions was merged into the GitHub repository before I moved the component around to work around the merge problem, so there are definitely not 2.5 weeks of missing contributions. https://github.com/tenacityteam/tenacity/commit/fada9fe851c4b3de01d9bf951e403f8583e9e66b is clearly still present in the repository for example, which is 16 days old.
